controllers/postgres_controller.go

```diff
@@ -214,53 +215,69 @@ func (r *PostgresReconciler) createZalandoPostgresql(ctx context.Context, z *pg.
 }
 
 // ensureZalandoDependencies makes sure Zalando resources are installed in the service-cluster.
-func (r *PostgresReconciler) ensureZalandoDependencies(ctx context.Context, pg *pg.Postgres) error {
-	namespace := pg.ToPeripheralResourceNamespace()
+func (r *PostgresReconciler) ensureZalandoDependencies(ctx context.Context, p *pg.Postgres) error {
+	namespace := p.ToPeripheralResourceNamespace()
 	isInstalled, err := r.IsOperatorInstalled(ctx, namespace)
 	if err != nil {
 		return fmt.Errorf("error while querying if zalando dependencies are installed: %v", err)
 	}
 
-	// TODO fetch secret
+	// fetch secret
+	backupSecret := &v1.Secret{}
+	backupNamespace := types.NamespacedName{
+		Name:      p.ToBackupSecretName(),
+		Namespace: p.Namespace,
+	}
+	if err := r.Client.Get(ctx, backupNamespace, backupSecret); err != nil {
+		return fmt.Errorf("error while getting the backup secret from control plane cluster: %v", err)
+	}
+	bucketName := "pg-" + p.Spec.ProjectID
+	awsEndpoint := string(backupSecret.Data[pg.BackupSecretS3BucketURL])
+	walES3Endpoint := "https+path://" + strings.TrimPrefix(awsEndpoint, "https://") // TODO find a cleaner way to do this...
```

Comment:
https://golang.org/pkg/net/url/#Parse gives you the URL which contains all parts of the url